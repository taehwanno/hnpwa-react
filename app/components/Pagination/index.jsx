import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import './Pagination.scss';

const propTypes = {
  currentPage: PropTypes.number,
  onPaginate: PropTypes.func,
};

const defaultProps = {
  currentPage: 1,
  onPaginate() {},
};

function Pagination({ currentPage, onPaginate }) {
  const prevDisabled = currentPage === 1;
  const nextDisabled = currentPage === 10;

  const prevButtonClassName = cx('Pagination__button', {
    'Pagination__button--disabled': prevDisabled,
  });

  const nextButtonClassName = cx('Pagination__button', {
    'Pagination__button--disabled': nextDisabled,
  });

  return (
    <div className="Pagination">
      <div className="Pagination__inner">
        <button
          className={prevButtonClassName}
          data-button="prev"
          disabled={prevDisabled}
          type="button"
          onClick={() => onPaginate(currentPage - 1)}
        >
          {'<'} Prev
        </button>
        {currentPage} / 10
        {' '}
        <button
          className={nextButtonClassName}
          data-button="next"
          disabled={nextDisabled}
          type="button"
          onClick={() => onPaginate(currentPage + 1)}
        >
          Next {'>'}
        </button>
      </div>
    </div>
  );
}

Pagination.propTypes = propTypes;
Pagination.defaultProps = defaultProps;

export default Pagination;
