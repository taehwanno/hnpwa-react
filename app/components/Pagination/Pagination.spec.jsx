import React from 'react';
import { shallow } from 'enzyme';

import Pagination from './';

describe('<Pagination />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have .Pagination__button--disabled when props.currentPage === 1', () => {
    const wrapper = shallow(<Pagination currentPage={1} />);
    const button = wrapper.find('.Pagination__button--disabled');
    expect(button.props()['data-button']).toBe('prev');
  });

  it('should have .Pagination__button--disabled when props.currentPage === 10', () => {
    const wrapper = shallow(<Pagination currentPage={10} />);
    const button = wrapper.find('.Pagination__button--disabled');
    expect(button.props()['data-button']).toBe('next');
  });

  it('should calls onPaginate with currentPage - 1 when simulate prev button click event', () => {
    const currentPage = 3;
    const onPaginate = jest.fn();
    const wrapper = shallow((
      <Pagination
        currentPage={currentPage}
        maxPage={10}
        onPaginate={onPaginate}
      />
    ));
    wrapper.find('.Pagination__button').at(0).simulate('click');
    expect(onPaginate).toHaveBeenCalledTimes(1);
    expect(onPaginate).toHaveBeenCalledWith(currentPage - 1);
  });

  it('should calls onPaginate with currentPage + 1 when simulate next button click event', () => {
    const currentPage = 3;
    const onPaginate = jest.fn();
    const wrapper = shallow((
      <Pagination
        currentPage={currentPage}
        maxPage={10}
        onPaginate={onPaginate}
      />
    ));
    wrapper.find('.Pagination__button').at(1).simulate('click');
    expect(onPaginate).toHaveBeenCalledTimes(1);
    expect(onPaginate).toHaveBeenCalledWith(currentPage + 1);
  });
});
