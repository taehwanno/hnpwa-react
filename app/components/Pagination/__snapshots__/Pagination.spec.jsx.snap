// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match snapshot when render default 1`] = `
<div
  className="Pagination"
>
  <div
    className="Pagination__inner"
  >
    <button
      className="Pagination__button Pagination__button--disabled"
      data-button="prev"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      &lt;
       Prev
    </button>
    1
     / 10
     
    <button
      className="Pagination__button"
      data-button="next"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Next 
      &gt;
    </button>
  </div>
</div>
`;
